UPtables/lib/asp/constraints/hard/mandatory_and_recommended_in_same_semester.lp

```diff
@@ -2,9 +2,18 @@
 %%% All courses that are mandatory for a given curriculum must be scheduled in different
 %%% periods if they are recommended for the same semester
 
-:- not { assigned(C,WD,TF) : recommendation(C,S,Cu), mandatory(C,Cu) } 1, semester(S), cu(Cu), weekday(WD,_), timeframe(TF,_).
+
+% not more than 2 mandatory lectures at one time (recommended in same semester)
+:- not { assigned(C,WD,TF) : recommendation(C,S,Cu), mandatory(C,Cu), course_component(C,_,_,_,_,lecture) } 1, semester(S), cu(Cu), weekday(WD,_), timeframe(TF,_).
+
+% not one mandatory lecture and more than one tutorial at one time (recommended in same semester)
```

Comment:
Sorry my comments are wrong:
- not: more than 1 mandatory lecture
- not: one mandatory and at least one tutorial